byway/src/main/webapp/home.js

```diff
@@ -44,7 +44,12 @@ function loadPastTrip(){
     tripIds.forEach(trip => {
```

Comment:
Notice that up here you're also trying to load every trip simultaneously, so it might still be possible to hit your rate limits unexpectedly. If that becomes a problem you may want to await up here as well.